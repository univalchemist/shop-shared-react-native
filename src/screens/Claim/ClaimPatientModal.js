import React from 'react';
import { connect } from 'react-redux';
import { ListPicker } from '@wrappers/components';
import PropTypes from 'prop-types';
import { change } from 'redux-form';
import { CLAIM_PATIENT_DETAILS } from '@routes';
import { categories } from '@store/analytics/trackingActions';

const ClaimPatientModal = ({
  change,
  userId,
  membersProfileOrder,
  unterminatedMembersMap,
  selectedPatientId,
  navigation,
}) => {
  if (!unterminatedMembersMap[userId].fullName) {
    return <></>;
  }
  const onPressItem = memberId => {
    const selectedItem = unterminatedMembersMap[memberId];
    change('claimDetailsForm', 'patientName', selectedItem.fullName);
    change('claimDetailsForm', 'selectedPatientId', memberId);
    navigation.navigate(CLAIM_PATIENT_DETAILS);
  };

  const data = membersProfileOrder
    .map(memberId => {
      const member = unterminatedMembersMap[memberId];
      if (!member) return null;
      return {
        key: memberId,
        value: member.fullName,
      };
    })
    .filter(item => !!item);

  const getActionParams = () => ({
    category: categories.CLAIMS_SUBMISSION,
    action: 'Select patient name',
  });

  return (
    <ListPicker
      data={data}
      onPressItem={onPressItem}
      selectedKey={selectedPatientId}
      getActionParams={getActionParams}
    />
  );
};

ClaimPatientModal.propTypes = {
  unterminatedMembersMap: PropTypes.shape({}),
  membersProfileOrder: PropTypes.array,
  selectedPatientId: PropTypes.string,
  change: PropTypes.func,
};

const mapStateToProps = ({
  user: { userId, membersProfileOrder, unterminatedMembersMap },
  form: {
    claimDetailsForm: { values },
  },
}) => ({
  userId,
  membersProfileOrder,
  unterminatedMembersMap,
  selectedPatientId: values?.selectedPatientId,
});

export default connect(mapStateToProps, { change })(ClaimPatientModal);
