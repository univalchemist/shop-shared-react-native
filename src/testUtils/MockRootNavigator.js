/* istanbul ignore file */
import React from 'react';
import { PUBLIC } from '@routes';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';

const Stack = createStackNavigator();

const MockRootNavigator = ({ component }) => {
  return (
    <NavigationContainer>
      <Stack.Navigator headerMode="none">
        <Stack.Screen name={PUBLIC}>{props => component}</Stack.Screen>
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default MockRootNavigator;
