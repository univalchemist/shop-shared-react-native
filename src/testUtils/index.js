/* istanbul ignore file */
import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { render as renderComponent } from 'react-native-testing-library';
import createMockStore from 'redux-mock-store';
import { ThemeProvider } from 'react-native-elements';
import { ThemeProvider as StyledThemeProvider } from 'styled-components/native';
import thunk from 'redux-thunk';
import { createPromise } from 'redux-promise-middleware';
import IntlWrapper from '@components/IntlWrapper';
import reducers from '@store/reducers';
import { mergeDeepRight } from 'ramda';
import theme from '../theme';
import messages from '../../messages/en-HK';
import healMessages from '@heal/messages/en-HK';
import { en_HK as shopMessages } from '@components/shops';

import MockRootNavigator from './MockRootNavigator';

const reduxPromiseMiddleware = createPromise({
  promiseTypeSuffixes: ['START', 'SUCCESS', 'ERROR'],
});

const defaultOptions = {
  initialState: {
    intl: {
      messages: { ...messages, ...healMessages, ...shopMessages },
      intlLocale: 'en-HK',
      momentLocale: 'en-gb',
      locale: 'en-HK',
      initialNow: Date.now(),
    },
    user: {},
  },
  api: {},
};

export const renderForTest = (
  Component,
  { initialState, api } = defaultOptions,
) => {
  const mockStore = configureMockStore(api);
  return renderComponent(
    <Provider store={mockStore(initialState)}>
      <IntlWrapper>
        <ThemeProvider>
          <StyledThemeProvider theme={theme}>
            <MockRootNavigator component={Component} />
          </StyledThemeProvider>
        </ThemeProvider>
      </IntlWrapper>
    </Provider>,
  );
};

export const configureMockStore = api => (initialState = {}) =>
  createMockStore([thunk.withExtraArgument({ api }), reduxPromiseMiddleware])(
    mergeDeepRight(defaultOptions.initialState, initialState),
  );

export const configureStore = (api, initialState) =>
  createStore(
    reducers,
    initialState,
    applyMiddleware(thunk.withExtraArgument({ api }), reduxPromiseMiddleware),
  );

export const render = (component, { initialState, api } = defaultOptions) => {
  const store = configureStore(
    { ...defaultOptions.api, ...api },
    { ...defaultOptions.initialState, ...initialState },
  );

  const Component = renderComponent(
    <Provider store={store}>
      <IntlWrapper>
        <ThemeProvider>
          <StyledThemeProvider theme={theme}>
            <MockRootNavigator component={component} />
          </StyledThemeProvider>
        </ThemeProvider>
      </IntlWrapper>
    </Provider>,
  );

  const rerender = () => {
    Component.rerender(
      <Provider store={store}>
        <IntlWrapper>
          <ThemeProvider>
            <StyledThemeProvider theme={theme}>
              <MockRootNavigator component={component} />
            </StyledThemeProvider>
          </ThemeProvider>
        </IntlWrapper>
      </Provider>,
    );
  };

  return [Component, store, rerender];
};

export const renderWithoutStack = (
  Component,
  { initialState, api } = defaultOptions,
) => {
  const mockStore = configureMockStore(api);
  return renderComponent(
    <Provider store={mockStore(initialState)}>
      <IntlWrapper>
        <ThemeProvider>
          <StyledThemeProvider theme={theme}>{Component}</StyledThemeProvider>
        </ThemeProvider>
      </IntlWrapper>
    </Provider>,
  );
};

export const renderForTestWithStore = render;
