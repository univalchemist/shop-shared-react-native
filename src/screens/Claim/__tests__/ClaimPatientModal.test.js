import React from 'react';
import ClaimPatientModal from '../ClaimPatientModal';
import {
  fireEvent,
  flushMicrotasksQueue,
  act,
} from 'react-native-testing-library';
import { change } from 'redux-form';
import { renderForTest, renderWithoutStack } from '../../../testUtils';

jest.mock('redux-form', () => ({
  change: jest.fn(() => ({ type: 'OK' })),
}));

const userWithoutFullname = {
  userId: '3',
  membersProfileOrder: ['3'],
  membersMap: {
    '3': {
      memberId: '3',
      fullName: '',
      relationships: null,
    },
    '1': {
      memberId: '1',
      fullName: '',
      relationships: null,
    },
  },
  unterminatedMembersMap: {
    '3': {
      memberId: '3',
      fullName: '',
      relationships: null,
    },
  },
};

const user = {
  userId: '3',
  membersProfileOrder: ['3', '27', '28'],
  membersMap: {
    '3': {
      memberId: '3',
      fullName: 'William Brown',
    },
    '27': {
      contactNumber: null,
      dateOfBirth: '1990-01-01T00:00:00',
      email: 'cath@test.com',
      externalId: '0000124',
      firstName: 'Catherine Brown',
      fullName: 'Catherine Brown Tan',
      gender: 'Female',
      lastName: 'Tan',
      memberId: '27',
      relationshipToEmployee: 'Spouse',
      role: 'Dependent',
    },
    '28': {
      contactNumber: null,
      dateOfBirth: '1990-01-01T00:00:00',
      email: 'george@test.com',
      externalId: '0000125',
      firstName: 'George Brown',
      fullName: 'George Brown Tan',
      gender: 'Male',
      lastName: 'Tan',
      memberId: '28',
      relationshipToEmployee: 'Child',
      role: 'Dependent',
    },
  },
  unterminatedMembersMap: {
    '3': {
      memberId: '3',
      fullName: 'William Brown',
    },
    '28': {
      contactNumber: null,
      dateOfBirth: '1990-01-01T00:00:00',
      email: 'george@test.com',
      externalId: '0000125',
      firstName: 'George Brown',
      fullName: 'George Brown Tan',
      gender: 'Male',
      lastName: 'Tan',
      memberId: '28',
      relationshipToEmployee: 'Child',
      role: 'Dependent',
    },
  },
};

const form = {
  claimDetailsForm: {
    values: { selectedPatientId: 0 },
  },
};

describe('ClaimPatientModal', () => {
  it('should render correctly', () => {
    const component = renderForTest(<ClaimPatientModal />, {
      initialState: { user, form },
    });
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should return null', () => {
    const component = renderWithoutStack(<ClaimPatientModal />, {
      initialState: { user: userWithoutFullname, form },
    });
    expect(component.toJSON()).toBeNull();
  });

  it('should indicate the selected patient', async () => {
    const component = renderForTest(<ClaimPatientModal />, {
      initialState: {
        user,
        form,
      },
    });
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('list item should fire form changes on press', async () => {
    const navigation = { navigate: jest.fn() };
    const component = renderForTest(
      <ClaimPatientModal navigation={navigation} />,
      {
        initialState: {
          user,
          form,
        },
      },
    );
    const listItem = component.getByText(user.membersMap[user.userId].fullName);
    await fireEvent.press(listItem);
    await flushMicrotasksQueue();

    expect(change).toBeCalledWith(
      'claimDetailsForm',
      'selectedPatientId',
      user.userId,
    );

    expect(change).toBeCalledWith(
      'claimDetailsForm',
      'patientName',
      user.membersMap[user.userId].fullName,
    );
  });

  it('list item should navigate back on press', async () => {
    const navigation = { navigate: jest.fn() };
    const component = renderForTest(
      <ClaimPatientModal navigation={navigation} />,
      {
        initialState: {
          user,
          form,
        },
      },
    );
    const listItem = component.getByText(user.membersMap[user.userId].fullName);
    act(() => {
      fireEvent.press(listItem);
    });
    await flushMicrotasksQueue();

    expect(navigation.navigate).toBeCalledTimes(1);
  });

  it('list item should update the form on press', async () => {
    const navigation = { navigate: jest.fn() };
    const component = renderForTest(
      <ClaimPatientModal navigation={navigation} />,
      {
        initialState: {
          user,
          form,
        },
      },
    );
    const listItem = component.getByText(user.membersMap[user.userId].fullName);
    await fireEvent.press(listItem);
    await flushMicrotasksQueue();

    expect(change).toBeCalledWith(
      'claimDetailsForm',
      'patientName',
      user.membersMap[user.userId].fullName,
    );
  });
});
