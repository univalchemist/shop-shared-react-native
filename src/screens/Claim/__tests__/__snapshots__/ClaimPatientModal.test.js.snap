// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaimPatientModal should indicate the selected patient 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        bg="gray.7"
                        flex={1}
                        onLayout={[Function]}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#f9f9f9",
                              "flexBasis": 0,
                              "flexGrow": 1,
                              "flexShrink": 1,
                            },
                          ]
                        }
                      >
                        <RCTScrollView
                          automaticallyAdjustContentInsets={false}
                          contentInset={
                            Object {
                              "bottom": 0,
                            }
                          }
                          enableAutomaticScroll={true}
                          enableOnAndroid={false}
                          enableResetScrollToCoords={true}
                          extraHeight={75}
                          extraScrollHeight={0}
                          getScrollResponder={[Function]}
                          handleOnScroll={[Function]}
                          innerRef={[Function]}
                          keyboardDismissMode="on-drag"
                          keyboardOpeningTime={250}
                          keyboardShouldPersistTaps="handled"
                          keyboardSpace={0}
                          onScroll={[Function]}
                          onScrollBeginDrag={null}
                          resetKeyboardSpace={[Function]}
                          scrollEventThrottle={16}
                          scrollForExtraHeightOnAndroid={[Function]}
                          scrollIndicatorInsets={
                            Object {
                              "right": 1,
                            }
                          }
                          scrollIntoView={[Function]}
                          scrollToEnd={[Function]}
                          scrollToFocusedInput={[Function]}
                          scrollToPosition={[Function]}
                          showsVerticalScrollIndicator={true}
                          update={[Function]}
                          viewIsInsideTabBar={false}
                        >
                          <View>
                            <RCTScrollView
                              ListFooterComponent={
                                <ForwardRef
                                  pb={24}
                                />
                              }
                              data={
                                Array [
                                  Object {
                                    "key": "3",
                                    "value": "William Brown",
                                  },
                                  Object {
                                    "key": "28",
                                    "value": "George Brown Tan",
                                  },
                                ]
                              }
                              disableVirtualization={false}
                              getItem={[Function]}
                              getItemCount={[Function]}
                              horizontal={false}
                              initialNumToRender={10}
                              keyExtractor={[Function]}
                              maxToRenderPerBatch={10}
                              numColumns={1}
                              onContentSizeChange={[Function]}
                              onEndReachedThreshold={2}
                              onLayout={[Function]}
                              onMomentumScrollEnd={[Function]}
                              onScroll={[Function]}
                              onScrollBeginDrag={[Function]}
                              onScrollEndDrag={[Function]}
                              removeClippedSubviews={false}
                              renderItem={[Function]}
                              scrollEventThrottle={50}
                              stickyHeaderIndices={Array []}
                              updateCellsBatchingPeriod={50}
                              viewabilityConfigCallbackPairs={Array []}
                              windowSize={21}
                            >
                              <View>
                                <View
                                  onLayout={[Function]}
                                  style={null}
                                >
                                  <View
                                    accessible={true}
                                    focusable={true}
                                    isTVSelectable={true}
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={null}
                                  >
                                    <View
                                      alignItems="center"
                                      flexDirection="row"
                                      justifyContent="space-between"
                                      style={
                                        Array [
                                          Object {
                                            "alignItems": "center",
                                            "backgroundColor": "#f9f9f9",
                                            "borderBottomColor": "#e1e1e1",
                                            "borderBottomWidth": 1,
                                            "display": "flex",
                                            "flexDirection": "row",
                                            "flexWrap": "wrap",
                                            "justifyContent": "space-between",
                                            "paddingBottom": 18,
                                            "paddingLeft": 32,
                                            "paddingRight": 32,
                                            "paddingTop": 18,
                                          },
                                          null,
                                        ]
                                      }
                                    >
                                      <View
                                        flex={1}
                                        style={
                                          Array [
                                            Object {
                                              "flexBasis": 0,
                                              "flexGrow": 1,
                                              "flexShrink": 1,
                                            },
                                          ]
                                        }
                                      >
                                        <Text
                                          color="gray.1"
                                          fontFamily="default"
                                          fontSize={16}
                                          letterSpacing={0.15}
                                          lineHeight={24}
                                          style={
                                            Array [
                                              Object {
                                                "color": "#646464",
                                                "fontFamily": "Arial",
                                                "fontSize": 16,
                                                "letterSpacing": 0.15,
                                                "lineHeight": 24,
                                              },
                                            ]
                                          }
                                        >
                                          William Brown
                                        </Text>
                                      </View>
                                    </View>
                                  </View>
                                </View>
                                <View
                                  onLayout={[Function]}
                                  style={null}
                                >
                                  <View
                                    accessible={true}
                                    focusable={true}
                                    isTVSelectable={true}
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={null}
                                  >
                                    <View
                                      alignItems="center"
                                      flexDirection="row"
                                      justifyContent="space-between"
                                      style={
                                        Array [
                                          Object {
                                            "alignItems": "center",
                                            "backgroundColor": "#f9f9f9",
                                            "borderBottomColor": "#e1e1e1",
                                            "borderBottomWidth": 1,
                                            "display": "flex",
                                            "flexDirection": "row",
                                            "flexWrap": "wrap",
                                            "justifyContent": "space-between",
                                            "paddingBottom": 18,
                                            "paddingLeft": 32,
                                            "paddingRight": 32,
                                            "paddingTop": 18,
                                          },
                                          null,
                                        ]
                                      }
                                    >
                                      <View
                                        flex={1}
                                        style={
                                          Array [
                                            Object {
                                              "flexBasis": 0,
                                              "flexGrow": 1,
                                              "flexShrink": 1,
                                            },
                                          ]
                                        }
                                      >
                                        <Text
                                          color="gray.1"
                                          fontFamily="default"
                                          fontSize={16}
                                          letterSpacing={0.15}
                                          lineHeight={24}
                                          style={
                                            Array [
                                              Object {
                                                "color": "#646464",
                                                "fontFamily": "Arial",
                                                "fontSize": 16,
                                                "letterSpacing": 0.15,
                                                "lineHeight": 24,
                                              },
                                            ]
                                          }
                                        >
                                          George Brown Tan
                                        </Text>
                                      </View>
                                    </View>
                                  </View>
                                </View>
                                <View
                                  onLayout={[Function]}
                                >
                                  <View
                                    pb={24}
                                    style={
                                      Array [
                                        Object {
                                          "paddingBottom": 24,
                                        },
                                      ]
                                    }
                                  />
                                </View>
                              </View>
                            </RCTScrollView>
                          </View>
                        </RCTScrollView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;

exports[`ClaimPatientModal should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        bg="gray.7"
                        flex={1}
                        onLayout={[Function]}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#f9f9f9",
                              "flexBasis": 0,
                              "flexGrow": 1,
                              "flexShrink": 1,
                            },
                          ]
                        }
                      >
                        <RCTScrollView
                          automaticallyAdjustContentInsets={false}
                          contentInset={
                            Object {
                              "bottom": 0,
                            }
                          }
                          enableAutomaticScroll={true}
                          enableOnAndroid={false}
                          enableResetScrollToCoords={true}
                          extraHeight={75}
                          extraScrollHeight={0}
                          getScrollResponder={[Function]}
                          handleOnScroll={[Function]}
                          innerRef={[Function]}
                          keyboardDismissMode="on-drag"
                          keyboardOpeningTime={250}
                          keyboardShouldPersistTaps="handled"
                          keyboardSpace={0}
                          onScroll={[Function]}
                          onScrollBeginDrag={null}
                          resetKeyboardSpace={[Function]}
                          scrollEventThrottle={16}
                          scrollForExtraHeightOnAndroid={[Function]}
                          scrollIndicatorInsets={
                            Object {
                              "right": 1,
                            }
                          }
                          scrollIntoView={[Function]}
                          scrollToEnd={[Function]}
                          scrollToFocusedInput={[Function]}
                          scrollToPosition={[Function]}
                          showsVerticalScrollIndicator={true}
                          update={[Function]}
                          viewIsInsideTabBar={false}
                        >
                          <View>
                            <RCTScrollView
                              ListFooterComponent={
                                <ForwardRef
                                  pb={24}
                                />
                              }
                              data={
                                Array [
                                  Object {
                                    "key": "3",
                                    "value": "William Brown",
                                  },
                                  Object {
                                    "key": "28",
                                    "value": "George Brown Tan",
                                  },
                                ]
                              }
                              disableVirtualization={false}
                              getItem={[Function]}
                              getItemCount={[Function]}
                              horizontal={false}
                              initialNumToRender={10}
                              keyExtractor={[Function]}
                              maxToRenderPerBatch={10}
                              numColumns={1}
                              onContentSizeChange={[Function]}
                              onEndReachedThreshold={2}
                              onLayout={[Function]}
                              onMomentumScrollEnd={[Function]}
                              onScroll={[Function]}
                              onScrollBeginDrag={[Function]}
                              onScrollEndDrag={[Function]}
                              removeClippedSubviews={false}
                              renderItem={[Function]}
                              scrollEventThrottle={50}
                              stickyHeaderIndices={Array []}
                              updateCellsBatchingPeriod={50}
                              viewabilityConfigCallbackPairs={Array []}
                              windowSize={21}
                            >
                              <View>
                                <View
                                  onLayout={[Function]}
                                  style={null}
                                >
                                  <View
                                    accessible={true}
                                    focusable={true}
                                    isTVSelectable={true}
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={null}
                                  >
                                    <View
                                      alignItems="center"
                                      flexDirection="row"
                                      justifyContent="space-between"
                                      style={
                                        Array [
                                          Object {
                                            "alignItems": "center",
                                            "backgroundColor": "#f9f9f9",
                                            "borderBottomColor": "#e1e1e1",
                                            "borderBottomWidth": 1,
                                            "display": "flex",
                                            "flexDirection": "row",
                                            "flexWrap": "wrap",
                                            "justifyContent": "space-between",
                                            "paddingBottom": 18,
                                            "paddingLeft": 32,
                                            "paddingRight": 32,
                                            "paddingTop": 18,
                                          },
                                          null,
                                        ]
                                      }
                                    >
                                      <View
                                        flex={1}
                                        style={
                                          Array [
                                            Object {
                                              "flexBasis": 0,
                                              "flexGrow": 1,
                                              "flexShrink": 1,
                                            },
                                          ]
                                        }
                                      >
                                        <Text
                                          color="gray.1"
                                          fontFamily="default"
                                          fontSize={16}
                                          letterSpacing={0.15}
                                          lineHeight={24}
                                          style={
                                            Array [
                                              Object {
                                                "color": "#646464",
                                                "fontFamily": "Arial",
                                                "fontSize": 16,
                                                "letterSpacing": 0.15,
                                                "lineHeight": 24,
                                              },
                                            ]
                                          }
                                        >
                                          William Brown
                                        </Text>
                                      </View>
                                    </View>
                                  </View>
                                </View>
                                <View
                                  onLayout={[Function]}
                                  style={null}
                                >
                                  <View
                                    accessible={true}
                                    focusable={true}
                                    isTVSelectable={true}
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={null}
                                  >
                                    <View
                                      alignItems="center"
                                      flexDirection="row"
                                      justifyContent="space-between"
                                      style={
                                        Array [
                                          Object {
                                            "alignItems": "center",
                                            "backgroundColor": "#f9f9f9",
                                            "borderBottomColor": "#e1e1e1",
                                            "borderBottomWidth": 1,
                                            "display": "flex",
                                            "flexDirection": "row",
                                            "flexWrap": "wrap",
                                            "justifyContent": "space-between",
                                            "paddingBottom": 18,
                                            "paddingLeft": 32,
                                            "paddingRight": 32,
                                            "paddingTop": 18,
                                          },
                                          null,
                                        ]
                                      }
                                    >
                                      <View
                                        flex={1}
                                        style={
                                          Array [
                                            Object {
                                              "flexBasis": 0,
                                              "flexGrow": 1,
                                              "flexShrink": 1,
                                            },
                                          ]
                                        }
                                      >
                                        <Text
                                          color="gray.1"
                                          fontFamily="default"
                                          fontSize={16}
                                          letterSpacing={0.15}
                                          lineHeight={24}
                                          style={
                                            Array [
                                              Object {
                                                "color": "#646464",
                                                "fontFamily": "Arial",
                                                "fontSize": 16,
                                                "letterSpacing": 0.15,
                                                "lineHeight": 24,
                                              },
                                            ]
                                          }
                                        >
                                          George Brown Tan
                                        </Text>
                                      </View>
                                    </View>
                                  </View>
                                </View>
                                <View
                                  onLayout={[Function]}
                                >
                                  <View
                                    pb={24}
                                    style={
                                      Array [
                                        Object {
                                          "paddingBottom": 24,
                                        },
                                      ]
                                    }
                                  />
                                </View>
                              </View>
                            </RCTScrollView>
                          </View>
                        </RCTScrollView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
